### PR TITLE
INTYGFV-15841: Added sessionId and traceId to monitorlogging.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

File: web/src/main/java/se/inera/intyg/minaintyg/common/filter/MDCServletFilter.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public class MDCServletFilter implements Filter {
+
+  private final MDCHelper mdcHelper;
+  @Value("${mdc.session.info.key}")
+  private String sessionInfoKey;
+  @Value("${mdc.trace.id.key}")
+  private String traceIdKey;
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/minaintyg/common/filter/MDCServletFilter.java:29]: Could be unit tested?

File: web/src/main/java/se/inera/intyg/minaintyg/common/filter/MDCServletFilter.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      throws IOException, ServletException {
+    if (verifyRequestType(request)) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/minaintyg/common/filter/MDCServletFilter.java:31]: For readability, consider changing the name of the method. It verifies that it is a HttpServletRequest.

Actually, you could instead use some standard java to make this happen (thanks to java17!)

    if (request instanceof HttpServletRequest http) {
      MDC.put(sessionInfoKey, mdcHelper.buildSessionInfo(http));
      MDC.put(traceIdKey, mdcHelper.buildTraceId(http));
    }

File: web/src/main/java/se/inera/intyg/minaintyg/common/filter/MDCServletFilter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      final var http = (HttpServletRequest) request;
+      MDC.put(sessionInfoKey, mdcHelper.buildSessionInfo(http));
+      MDC.put(traceIdKey, mdcHelper.buildTraceId(http));
+    }
+    try {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/minaintyg/common/filter/MDCServletFilter.java:36]: Should the try-statement cover the whole method body to make sure the MDC is cleared no matter what unexpected things happens?

File: web/src/main/java/se/inera/intyg/minaintyg/common/filter/MDCHelper.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    return request instanceof HttpServletRequest;
+  }
+
+  private String getSessionIdFromCookie(HttpServletRequest http) {
+    final var cookies = http.getCookies();
+    if (cookies == null) {
+      return null;
+    }
+    return Stream.of(cookies)
+        .filter(cookie -> "SESSION".equals(cookie.getName()))
```
[the remark this answers is not in the document]
Comment (ChrilleSigerhed) [reply web/src/main/java/se/inera/intyg/minaintyg/common/filter/MDCHelper.java:33]: Did not contain a spring session.

image

Could this be related to the request beeing done from Postman?

**Review by mww-sopra-steria — APPROVED** [review]
Nice! 

Do you mind adding the trace header to the intygstjanst integration as well? Now when it has been merged.

**Pull request merged**